### ISSUE TITLE
Deleting patterns causes screen to go blank/black

### ISSUE THREAD

**Issue by maddisondesigns** [WordPress/gutenberg#64220]
### Description

Tried to delete two Synced patterns. After selecting both and then clicking delete from the Edit menu, the screen went completely blank/black.

![Deleting patterns](https://github.com/user-attachments/assets/08dc7575-522c-45eb-a17f-ab3a8ad26331)

After navigating back to my page, I then created another pattern and then tried deleting it and the exact same thing happened.

The following 4 errors appear in the Console after clicking delete

```
Uncaught TypeError: item is undefined
    getItemTitle http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:25237
    RenderModal http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:25277
    React 17
    onStoreChange http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:4053
    onChange http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:4059
    unsubscribeFromStore http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:3374
    instantiate http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:2828
    dispatch http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:968
    createThunkMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1710
    createMiddleware Redux
    promiseMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1656
    createResolversCacheMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1700
    apply http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:2735
    deleteEntityRecord http://your-wordpress-site.com/wp-includes/js/dist/core-data.js?ver=1b304f4025915ff3c84b:2191
    createThunkMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1708
    createMiddleware Redux
    promiseMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1656
    createResolversCacheMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1700
    bindAction http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:2721
    promiseResult http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:5515
    removeTemplates http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:5514
    createThunkMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1708
    createMiddleware Redux
    promiseMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1656
    createResolversCacheMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1700
    bindAction http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:2721
    onClick http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:25293
    key http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:2265
    React 32
    onClick http://your-wordpress-site.com/wp-includes/js/dist/edit-site.js?ver=be992a2a06c05b011294:25163
    onClick http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:72265
    useEvent http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:2651
    React 9
editor.js:25237:7

====================================================================================

Uncaught TypeError: item is undefined
    getItemTitle http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:25237
    RenderModal http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:25277
    React 18
    onStoreChange http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:4053
    onChange http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:4059
    unsubscribeFromStore http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:3374
    instantiate http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:2828
    dispatch http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:968
    createThunkMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1710
    createMiddleware Redux
    promiseMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1656
    createResolversCacheMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1700
    apply http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:2735
    deleteEntityRecord http://your-wordpress-site.com/wp-includes/js/dist/core-data.js?ver=1b304f4025915ff3c84b:2191
    createThunkMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1708
    createMiddleware Redux
    promiseMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1656
    createResolversCacheMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1700
    bindAction http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:2721
    promiseResult http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:5515
    removeTemplates http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:5514
    createThunkMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1708
    createMiddleware Redux
    promiseMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1656
    createResolversCacheMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1700
    bindAction http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:2721
    onClick http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:25293
    key http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:2265
    React 32
    onClick http://your-wordpress-site.com/wp-includes/js/dist/edit-site.js?ver=be992a2a06c05b011294:25163
    onClick http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:72265
    useEvent http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:2651
    React 9
editor.js:25237:7

====================================================================================

The above error occurred in the <RenderModal> component:

RenderModal@http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:25260:16
div
div
div
StyleProvider@http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:33537:7
div
UnforwardedModal@http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:47010:7
bulk_actions_ActionWithModal@http://your-wordpress-site.com/wp-includes/js/dist/edit-site.js?ver=be992a2a06c05b011294:25120:38
BulkActions@http://your-wordpress-site.com/wp-includes/js/dist/edit-site.js?ver=be992a2a06c05b011294:25187:21
div
withEmotionCache/<@http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:12988:47
UnforwardedView@http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:13877:25
UnconnectedHStack@http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:32135:32
div
DataViews@http://your-wordpress-site.com/wp-includes/js/dist/edit-site.js?ver=be992a2a06c05b011294:32482:19
div
div
NavigableRegion@http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:7496:25
Page@http://your-wordpress-site.com/wp-includes/js/dist/edit-site.js?ver=be992a2a06c05b011294:32638:14
BlockRefsProvider@http://your-wordpress-site.com/wp-includes/js/dist/block-editor.js?ver=2cb36d67922bfdc27a1d:19803:27
Provider@http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:33904:18
ExperimentalBlockEditorProvider<@http://your-wordpress-site.com/wp-includes/js/dist/block-editor.js?ver=2cb36d67922bfdc27a1d:41832:7
withRegistryProvider</<@http://your-wordpress-site.com/wp-includes/js/dist/block-editor.js?ver=2cb36d67922bfdc27a1d:41366:117
DataviewsPatterns@http://your-wordpress-site.com/wp-includes/js/dist/edit-site.js?ver=be992a2a06c05b011294:35096:7
div
div
div
Layout@http://your-wordpress-site.com/wp-includes/js/dist/edit-site.js?ver=be992a2a06c05b011294:40334:3
RouterProvider@http://your-wordpress-site.com/wp-includes/js/dist/router.js?ver=bce4a12305810093635e:936:24
GlobalStylesProvider@http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:19418:30
SlotFillProvider@http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:33742:26
provider_SlotFillProvider@http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:33840:35
Provider@http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:33904:18
App@http://your-wordpress-site.com/wp-includes/js/dist/edit-site.js?ver=be992a2a06c05b011294:40524:55

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries. react-dom.development.js:18704:15

====================================================================================

Uncaught TypeError: item is undefined
    getItemTitle http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:25237
    RenderModal http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:25277
    React 15
    onStoreChange http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:4053
    onChange http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:4059
    unsubscribeFromStore http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:3374
    instantiate http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:2828
    dispatch http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:968
    createThunkMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1710
    createMiddleware Redux
    promiseMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1656
    createResolversCacheMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1700
    apply http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:2735
    deleteEntityRecord http://your-wordpress-site.com/wp-includes/js/dist/core-data.js?ver=1b304f4025915ff3c84b:2191
    createThunkMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1708
    createMiddleware Redux
    promiseMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1656
    createResolversCacheMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1700
    bindAction http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:2721
    promiseResult http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:5515
    removeTemplates http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:5514
    createThunkMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1708
    createMiddleware Redux
    promiseMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1656
    createResolversCacheMiddleware http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:1700
    bindAction http://your-wordpress-site.com/wp-includes/js/dist/data.js?ver=6f178ef23ef546c51cae:2721
    onClick http://your-wordpress-site.com/wp-includes/js/dist/editor.js?ver=e15506639990b108fc28:25293
    key http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:2265
    React 32
    onClick http://your-wordpress-site.com/wp-includes/js/dist/edit-site.js?ver=be992a2a06c05b011294:25163
    onClick http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:72265
    useEvent http://your-wordpress-site.com/wp-includes/js/dist/components.js?ver=54eab95952fe254d64e9:2651
    React 9
editor.js:25237:7
```

(Site is on my local dev server)

### Step-by-step reproduction instructions

- Create Synced Pattern
- Select Manage Patterns from the Options Menu
- On the manage patterns screen, select your pattern and then click delete from the Edit menu
- Watch screen go black

### Environment info

**WP:** 6.6.1
**Firefox:** 128.0.3
**macOS:** Sonoma 14.5

No plugins activated.

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by Mamaduka** [comment]
Thanks, @maddisondesigns!

I can reproduce the same bug with Template Parts. It only happens when using the Edit menu; deletion works correctly from quick actions.

**Comment by Mamaduka** [comment]
Fixed via #64250.
